### PR TITLE
DOC: add examples in distance for correlation, num_obs_dm and num_obs_y

### PR DESCRIPTION
**Pull request by SoubhagyaKhan**
Added examples section to correlation, num_obs_dm and num_obs_y functions as per issue #7168

#### Reference issue

#7168 
#### What does this implement/fix?

Added functions to correlation, num_obs_dm and num_obs_y function

### CODE REVIEW

**Review by tupui — CHANGES_REQUESTED** [review]
Hi @SoubhagyaKhan, thank you for the PR. There are a few things to adjust like: PEP8, linter and also since these are examples we should describe a bit what we are doing and why.

For the linters you can check locally using `dev.py`. e.g. for the linter:

```
python dev.py lint
```

Please have a look at our contributing guide for more detailed info. Or feel free to ask questions 😃 https://scipy.github.io/devdocs/dev/contributor/development_workflow.html

Also please add the following in your commit message: `[skip actions] [skip cirrus]`

File: scipy/spatial/distance.py [hunk 1]
```diff
@@ -603,6 +603,16 @@ def correlation(u, v, w=None, centered=True):
...
 
+    Examples
+    --------
+    # To find the correlation distance between two arrays(length of each array must be same) without weights:
+    >>> from scipy.spatial import distance
+    >>> distance.correlation([1, 0, 1], [2, 3, 1])
+    1.8660245037844386
+    # To find the correlation distance between two arrays with weights provided:
+    >>> distance.correlation([1, 0, 1], [2, 2, 1], [2, 3, 1])
+    1.4472135954999579
```
Comment (j-bowhay) [inline scipy/spatial/distance.py:614]:
```suggestion
    Examples
    --------
    To find the correlation distance between two arrays (length of each
    array must be same) without weights:
    
    >>> from scipy.spatial import distance
    >>> distance.correlation([1, 0, 1], [2, 3, 1])
    1.8660245037844386
    
    To find the correlation distance between two arrays with weights provided:
    
    >>> distance.correlation([1, 0, 1], [2, 2, 1], [2, 3, 1])
    1.4472135954999579
```

File: scipy/spatial/distance.py [hunk 2]
```diff
@@ -2574,7 +2584,12 @@ def num_obs_dm(d):
     -------
     num_obs_dm : int
         The number of observations in the redundant distance matrix.
-
+    Examples
+    --------
+    # To compute the number of observations that correspond to a square matrix:
+    >>> from scipy.spatial import distance
+    >>> distance.num_obs_dm([[2, 4, 5], [4, 5, 6], [7, 8, 1]])  # Argument must be a square matrix
+    3
```
Comment (j-bowhay) [inline scipy/spatial/distance.py:2592]:
```suggestion
    Examples
    --------
    To compute the number of observations that correspond to a square matrix:
    
    >>> from scipy.spatial import distance
    >>> distance.num_obs_dm([[2, 4, 5], [4, 5, 6], [7, 8, 1]])  # Argument must be a square matrix
    3
```

File: scipy/spatial/distance.py [hunk 3]
```diff
@@ -2595,7 +2610,14 @@ def num_obs_y(Y):
...
         The number of observations in the condensed distance matrix `Y`.
-
+    
+    Examples
+    --------
+    # To find the number of obersvations in a condensed distance matrix,here a one-dimensional array:
+    
+    >>> from scipy.spatial import distance
+    >>> distance.num_obs_y([1, 2, 3])  # Argument must be one-dimensional array
+    3
```
Comment (j-bowhay) [inline scipy/spatial/distance.py:2620]:
```suggestion
    Examples
    --------
    To find the number of observations in a condensed distance matrix,
    here a one-dimensional array:
    
    >>> from scipy.spatial import distance
    >>> distance.num_obs_y([1, 2, 3])
    3
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you for updating the PR.

Please test locally the changes. If you accept some suggestions edit the commit message (both to have meaningful messages and also skip CI). Lastly, it's better to accept suggestions in batches you can do that from the file changed tab. This limits the notifications we get.

**Comment by SoubhagyaKhan** [comment]
Ok,Sir!Thank you for the advice.Will changing from 'weights' to 'weight'
resolve the issue?

File: scipy/spatial/distance.py [hunk 4]
```diff
@@ -603,6 +603,20 @@ def correlation(u, v, w=None, centered=True):
...
+    To find the correlation distance between two arrays (length of each 
+    array must be same) without weights:
+    
+    >>> from scipy.spatial import distance
+    >>> distance.correlation([1, 0, 1], [2, 3, 1])
+    1.8660245037844386
+    
+    To find the correlation distance between two arrays with weights provided:
+    
+    >>> distance.correlation([1, 0, 1], [2, 2, 1], [2, 3, 1])
```
[the remark this answers is not in the document]
Comment (tupui) [reply scipy/spatial/distance.py:617]: Please don't resolve discussions. It helps reviewers to know what they asked.

It's also not resolved as what I asked was to use a the keyword in the code like `w=...`

**Review by tupui — COMMENTED** [review]
Please try to follow the suggested workflow. I know it's a bit tedious at first but there are good reasons for it 😃 (commit naming, skipping CI, suggestion, local testing)

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you for the update. It looks close to it in terms of formatting. Though I am still missing the value these examples would bring. I am personally still missing a bit of explanations of the value these functions bring. i.e. just running the function with random inputs is not helping a lot newcomers to understand a function.

**Comment by j-bowhay** [comment]
> Thank you for the update. It looks close to it in terms of formatting. Though I am still missing the value these examples would bring. I am personally still missing a bit of explanations of the value these functions bring. i.e. just running the function with random inputs is not helping a lot newcomers to understand a function.

For example, for correlation distance demonstrating some of the properties from the [Wikipedia article](https://en.wikipedia.org/wiki/Distance_correlation#Distance_correlation_2) could be nice

**Comment by SoubhagyaKhan** [comment]
Reviewers @tupui, @j-bowhay any more changes required ?If so, please tell me!

**Comment by tupui** [comment]
@SoubhagyaKhan I am still missing some explanations on what we are doing in correlation. What are these values, how to interpret the result, etc.

**Comment by j-bowhay** [comment]
Please have a look at https://numpydoc.readthedocs.io/en/v1.5.0/format.html#sections and the order sections need to go in, thanks!

**Comment by melissawm** [comment]
Hello @SoubhagyaKhan :wave: would you be willing to address the comments above? Thanks!

**Comment by j-bowhay** [comment]
Thanks @SoubhagyaKhan for your contribution unfortunately an example has now been added for this function but this is appreciated regardless

**Pull request closed without merging**